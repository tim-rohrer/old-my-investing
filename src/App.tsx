import React from "react";
import { ThemeProvider, StyledEngineProvider } from "@mui/material/styles";
import { Backdrop, CircularProgress } from "@mui/material";
import Home from "./features/home/Home";
import theme from "./common/theme";
import { selectAppIsThinking } from "./features/system/systemSlice";
import { useSelector } from "react-redux";


// declare module "@mui/styles/defaultTheme" {
//   // eslint-disable-next-line @typescript-eslint/no-empty-interface
//   interface DefaultTheme extends Theme {}
// }

const App: React.FC<Record<any, any>> = () => {
  // const classes = useStyles();
  const isAppThinking = useSelector(selectAppIsThinking);

  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <Home title="My App (Prototype)"/>
          <Backdrop 
            // eslint-disable-next-line @typescript-eslint/no-shadow
            sx={{ color: "#fff", zIndex: (theme) => theme.zIndex.drawer + 1 }}
            open={isAppThinking}>
            <CircularProgress color="inherit" />
          </Backdrop>
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

export default App;
