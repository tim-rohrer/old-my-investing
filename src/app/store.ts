import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import counterReducer from "../features/counter/counterSlice";
import systemReducer from "../features/system/systemSlice";
import securitiesReducer from "../features/securities/securitiesSlice";
import companyProfilesReducer from "../features/companyAnalysis/companyProfilesSlice";
import portfolioReducer from "../features/portfolio/portfolioSlice";
import quickenConnectorReducer from "../features/quickenConnector/quickenConnectorSlice";

export const reducer = {
  counter: counterReducer,
  system: systemReducer,
  securities: securitiesReducer,
  companyProfiles: companyProfilesReducer,
  portfolio: portfolioReducer,
  quickenConnector: quickenConnectorReducer,
};

export const store = configureStore({
  reducer,
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
