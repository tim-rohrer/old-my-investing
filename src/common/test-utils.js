// test-utils.js
import React from "react";
// import { render as rtlRender } from '@testing-library/react';
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { createStore } from "redux";
import { Provider } from "react-redux";
import {
  initialStateOriginal as reducerInitialState,
  reducer,
  initialStateOriginal,
} from "../app/store";

// const store = configureStore({ reducer });

export function renderWithRedux(
  ui,
  {
    initialState = initialStateOriginal,
    store = configureStore({ reducer }),
  } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}
// export function customRender(
//   ui,
//   {
//     initialState: RootState = reducerInitialState,
//     rtlStore,
//     ...renderOptions
//   } = {}
// ) {
//   function Wrapper({ children }) {
//     const rtlStore = configureStore({ reducer });
//     return <Provider store={rtlStore}>{children}</Provider>;
//   }
//   return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// }

// re-export everything
export * from "@testing-library/react";

// override render method
// export { customRender as render };
