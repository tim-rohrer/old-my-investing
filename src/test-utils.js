// test-utils.js
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
// import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { reducer } from './app/store'
// import { initialState as reducerInitialState, reducer } from './reducer'
const reducerInitialState = {}

function render(
  ui,
  {
    initialState: RootState = reducerInitialState,
    rtlStore,
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const rtlStore = configureStore({ reducer })
    return <Provider store={rtlStore}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'

// override render method
export { render }