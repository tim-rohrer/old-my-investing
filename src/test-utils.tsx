// test-utils.tsx
import * as React from "react"
import { Provider } from "react-redux"
import { createStore } from "redux"

import { queries, Queries } from "@testing-library/dom"
import { render as rtlRender, RenderOptions, RenderResult } from "@testing-library/react"

import { rootReducer, State, Store } from "./app/store"

type ExtraOptions = {
  initialState?: State;
  store?: Store;
};

const render = <
  Q extends Queries = typeof queries,
  C extends Element | DocumentFragment = HTMLElement
>(
  ui: React.ReactElement,
  options: ExtraOptions & RenderOptions<Q, C> = {} //:
): RenderResult<Q, C> & { store: Store } => {
  const {
    initialState = undefined, // reducerInitialState,
    store = createStore(rootReducer, initialState),
    ...renderOptions
  } = options;

  type Props = {
    children?: React.ReactNode
  }

  const Wrapper = ({ children }: Props) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }),
    store,
  };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
