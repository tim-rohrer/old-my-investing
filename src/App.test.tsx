import React from "react";
import { render, cleanup } from "./test-utils";
// import '@testing-library/jest-dom/extend-expect'
import App from "./App";
import { Provider } from "react-redux";
import { store } from "./app/store";

afterEach(cleanup);

describe("App", () => {
  it("renders a snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders my title", () => {
    const { getByText } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(getByText(/Investing/i)).toBeInTheDocument();
  });
});
