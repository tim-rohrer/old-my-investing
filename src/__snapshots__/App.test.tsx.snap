// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-noWrap"
          >
            My Investing (Prototype)
          </h6>
          <div
            class="makeStyles-search-4"
          >
            <div
              class="makeStyles-searchIcon-5"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root makeStyles-inputRoot-6"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input makeStyles-inputInput-7"
                placeholder="Symbol…"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </header>
      <div
        class="CompanyProfile"
      >
        <header
          class="CompanyProfile-header"
        >
          <h1>
            Company Profile
          </h1>
          <form>
            <label
              for="tickerInput"
            >
              <strong>
                Ticker: 
              </strong>
            </label>
             
            <input
              id="tickerInput"
              type="text"
              value=""
            />
             
          </form>
        </header>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-backdrop-10"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
